src/components/Calculator.js

```diff
@@ -1,39 +1,51 @@
-import React from 'react';
+import React, { useState } from 'react';
 import '../App.css';
-import ChildResult from './ChildResult';
+import calculate from '../logic/calculate';
 
 function Calculator() {
+  const [state, setState] = useState({ total: 0, next: null, operation: null });
+
+  const { total, next, operation } = state;
+
+  const calculateResult = (calculateButton) => {
+    setState((state) => calculate(state, calculateButton.target.textContent));
+  };
+
   return (
     <div className="container flex">
-      <ChildResult />
+      <div className="child result flex">
+        {total}
+        {operation}
+        {next}
+      </div>
       <div className="child flex">
-        <button type="button" className="button key">AC</button>
-        <button type="button" className="button key">÷/-</button>
-        <button type="button" className="button key">%</button>
-        <button type="button" className="button operator">÷</button>
+        <button type="button" className="button key" onClick={calculateResult}>AC</button>
+        <button type="button" className="button key" onClick={calculateResult}>÷/-</button>
```

Comment:
- [ ] Kindly make sure that the `+/-`  button is working properly. As the buttons in this project work using the text that's written in them, using a different text or having a typo could result in unwanted behavior.